### PR TITLE
Ethereum coding guide content from substrate style guide

### PR DESCRIPTION
**Pull request by joshua-mir**
Maybe we should just link to the substrate style guide instead?

### CODE REVIEW

File: Coding-guide.md [hunk 1]
```diff
@@ -2,23 +2,128 @@
 title: Coding Guide
```
Comment (ltfschoen) [inline Coding-guide.md]: I'd suggest renaming the title to "Style Guide for Rust in Parity Ethereum" so it's consistent with https://github.com/paritytech/wiki/blob/master/Substrate-Style-Guide.md

Comment (ltfschoen) [reply Coding-guide.md]: I'd also suggest renaming the file to Parity-Ethereum-Style-Guide.md

Comment (joshua-mir) [reply Coding-guide.md]: Agree, just need to update the sidebar and any links now 👍

Comment (ltfschoen) [reply Coding-guide.md]: Yes, we still need to update the following:
* paritytech/wiki/Parity-Ethereum.md - change `[Coding guide](Coding-guide)` to `[Parity Ethereum Style Guide](Parity-Ethereum-Style-Guide)`
* paritytech/wiki/_includes/_Sidebar.md - change `[Coding Guide](Coding-guide)` to `[Parity Ethereum Style Guide](Parity-Ethereum-Style-Guide)`

I can't think of anywhere else that links to it...

**Comment by ltfschoen** [comment]
@joshua-mir are you going to make the changes mentioned in https://github.com/paritytech/wiki/pull/332#discussion_r283916195 in this PR?

**Comment by joshua-mir** [comment]
Yep

**Pull request merged**